File: Programmers/javascript/hanghae99_4th/q33_로또의_최고_순위와_최저 순위.js
function solution(lottos, win_nums) {
    let answer = [];
    let matchingNum = 0;
    let countZero = 0;

    for (let i = 0; i < lottos.length; i++) {
        if (lottos[i] == 0) countZero++;
        if (win_nums.includes(lottos[i])) matchingNum++;
    }
    answer.push(getRate(matchingNum + countZero));
    answer.push(getRate(matchingNum));
    return answer;
}

function getRate(count) {
    if (count >= 2 && count <= 6) return 7 - count;
    else return 6;
}

console.log(solution([44, 1, 0, 0, 31, 25], [31, 10, 45, 1, 6, 19]));

File: Programmers/python/prac/q33_로또의_최고_순위와_최저 순위.py
def solution(lottos, win_nums):
    temp = [n for n in lottos if n in win_nums]
    max = 7 - (lottos.count(0)+len(temp)) if (lottos.count(0)+len(temp)) >= 1 else 6
    min = 7 - len(temp) if len(temp) >= 1 else 6
    return [max, min]
